Failed to build ESBMC

Hi, I'm not able to build ESBMC due to the error message:

```
[cmake] CMake Error at src/c2goto/CMakeLists.txt:32 (cmake_language):
[cmake]   Unknown CMake command "cmake_language".
[cmake] Call Stack (most recent call first):
[cmake]   src/c2goto/CMakeLists.txt:47 (foreach_in_cross_product_rec)
[cmake]   src/c2goto/CMakeLists.txt:47 (foreach_in_cross_product_rec)
[cmake]   src/c2goto/CMakeLists.txt:47 (foreach_in_cross_product_rec)
[cmake]   src/c2goto/CMakeLists.txt:53 (foreach_in_cross_product_rec)
[cmake]   src/c2goto/CMakeLists.txt:82 (foreach_in_cross_product)
[cmake] 
[cmake] 
[cmake] -- Configuring incomplete, errors occurred!
[cmake] See also "/home/kunjian/ESBMC_Project/esbmc/build/CMakeFiles/CMakeOutput.log".
[cmake] See also "/home/kunjian/ESBMC_Project/esbmc/build/CMakeFiles/CMakeError.log".
[proc] The command: /usr/bin/cmake --no-warn-unused-cli -DClang_DIR=/home/kunjian/local/llvm15-debug -DLLVM_DIR=/home/kunjian/local/llvm15-debug -DZ3_DIR=/home/kunjian/ESBMC_Project/z3 -DENABLE_OLD_FRONTEND=On -DBUILD_TESTING=On -DBUILD_STATIC=Off -DCMAKE_BUILD_TYPE=Debug -DENABLE_REGRESSION=On -DENABLE_SOLIDITY_FRONTEND=On -DBoolector_DIR=/home/kunjian/ESBMC_Project/boolector-release -DENABLE_MATHSAT=ON -DMathsat_DIR=/home/kunjian/ESBMC_Project/mathsat -DENABLE_YICES=On -DYices_DIR=/home/kunjian/ESBMC_Project/yices -DCVC4_DIR=/home/kunjian/ESBMC_Project/cvc4 -DGMP_DIR=/home/kunjian/ESBMC_Project/gmp -DBitwuzla_DIR=/home/kunjian/ESBMC_Project/bitwuzla-release -DENABLE_COVERAGE=Off -DCMAKE_EXPORT_COMPILE_COMMANDS:BOOL=TRUE -DCMAKE_BUILD_TYPE:STRING=Debug -DCMAKE_C_COMPILER:FILEPATH=/usr/bin/clang -DCMAKE_CXX_COMPILER:FILEPATH=/usr/bin/clang++ -S/home/kunjian/ESBMC_Project/esbmc -B/home/kunjian/ESBMC_Project/esbmc/build -G Ninja exited with code: 1
```

Q1. Does it also happen to other people? or is it just me?

Two more related questions as I just noticed that there were some changes in BUILDING.md...

Q2. Is clang-11 still supported? I'm still using clang-11.0.0 since the dependency says
![Screenshot 2023-07-10 at 08 00 02](https://github.com/esbmc/esbmc/assets/32592800/6929023a-99ee-478d-8374-ee063a33697e)

Q3. Are we going to use CHERI-clang-13 instead of the standard clang-13 for future build? 
![Screenshot 2023-07-10 at 08 03 15](https://github.com/esbmc/esbmc/assets/32592800/15753998-5dd1-41e0-ac09-6fee17031ac2)

--- a/BUILDING.md
+++ b/BUILDING.md
@@ -14,6 +14,7 @@ Before starting, note that ESBMC is mainly distributed under the terms of the [A
 |-----------|----------|-----------------|
 | clang     | yes      | 11.0.0          |
 | boost     | yes      | 1.77            |
+| CMake     | yes      | 3.18.0          |
 | Boolector | no       | 3.2.2           |
 | CVC4      | no       | 1.8             |
 | MathSAT   | no       | 5.5.4           |

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -1,7 +1,7 @@
 #################################
 # Project Information           #
 #################################
-cmake_minimum_required (VERSION 3.7)
+cmake_minimum_required (VERSION 3.18)
 
 # HACK: This msvc policy needs to be setted before project()
 # This can be removed when we move to CMake >= 3.15

